components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/DisordersController.java

```diff
@@ -0,0 +1,228 @@
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.ListProperty;
+
+/**
+ * Handles the patients disorders.
+ *
+ * @version $Id$
+ * @since 1.3RC1
+ */
+@Component(roles = { PatientDataController.class })
+@Named("disorders")
+@Singleton
+public class DisordersController extends AbstractComplexController<Disorder>
+{
+    protected static final String JSON_KEY_DISORDERS = "disorders";
+
+    private static final String CONTROLLER_NAME = JSON_KEY_DISORDERS;
+
+    private static final String DISORDER_PROPERTIES_OMIMID = "omim_id";
+
+    private static final String[] DISORDER_PROPERTIES = new String[] { DISORDER_PROPERTIES_OMIMID };
+
+    @Inject
+    private Logger logger;
+
+    /** Provides access to the current execution context. */
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Override
+    public String getName()
+    {
+        return CONTROLLER_NAME;
+    }
+
+    @Override
+    protected String getJsonPropertyName()
+    {
+        return CONTROLLER_NAME;
+    }
+
+    @Override
+    protected List<String> getProperties()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected List<String> getBooleanFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected List<String> getCodeFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public IndexedPatientData<Disorder> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+            BaseObject data = doc.getXObject(Patient.CLASS_REFERENCE);
+            if (data == null) {
+                return null;
+            }
+
+            List<Disorder> disorders = new ArrayList<>();
+
+            for (String property : DISORDER_PROPERTIES) {
+                ListProperty values = (ListProperty) data.get(property);
+                if (values != null) {
+                    for (String value : values.getList()) {
+                        if (StringUtils.isNotBlank(value)) {
+                            disorders.add(new PhenoTipsDisorder(values, value));
+                        }
+                    }
+                }
+            }
+            if (disorders.isEmpty()) {
+                return null;
+            } else {
+                return new IndexedPatientData<>(getName(), disorders);
+            }
+        } catch (Exception e) {
+            this.logger.error("Failed to access patient data for [{}]: {}", patient.getDocument(), e.getMessage());
+        }
+        return null;
+    }
+
+    @Override
+    public void writeJSON(Patient patient, JSONObject json, Collection<String> selectedFieldNames)
+    {
+        if (selectedFieldNames != null && !selectedFieldNames.contains(DISORDER_PROPERTIES)) {
```

Comment:
Then why we have it like that here https://github.com/phenotips/phenotips/blob/master/components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatient.java#L355 ?